methods-react/src/List.js

```diff
@@ -1,50 +1,61 @@
-// import StyledInput from './styleditems/styledInput'
 import Input from './Input'
+//REDUX
+import { useSelector } from 'react-redux';
+import IncludeFilter from './const/includeFilter';
+import CompareFilter from './const/compareFilter';
 
 const  List = (props) => {
-  let {allMethods} = useContext(Context)
-  const [filteredData, setFilteredData] = useState(allMethods)
-  const [filteredDataToShow, setFilteredDataToShow] = useState(allMethods)
+  
+  const newList = useSelector((state) => state.methods);
+  
+  const [filteredDataToShow, setFilteredDataToShow] = useState(newList.value)
   const [filterValue, setFilterValue] = useState('')
   
-  const filterData = (evt) => {
+  const filterData =  (evt) => {
     setFilterValue(evt.target.value)
     let filter = evt.target.value.toLowerCase()
-    const runFilter = () => setFilteredData(allMethods.filter(item => item.name.toLowerCase().includes(filter)))
-    setFilteredDataToShow(allMethods.filter(item => item.name.toLowerCase().includes(filter)))
-    if (evt.target.value === '') {
-        runFilter()
-    } else {
-        setTimeout(() => {
-            if (filter === evt.target.value.toLowerCase()) {
-                runFilter()
-            }
-        }, 2000)
-    }
+    setFilteredDataToShow(IncludeFilter(newList.value, filter))
 }
   
   const resetFilter = () => {
     setFilterValue('')
-    setFilteredDataToShow(allMethods)
-    setFilteredData(allMethods)
+    setFilteredDataToShow(CompareFilter(newList.value, props.header))
   }
+    
+    const listTorender = filterValue !=='' ? filteredDataToShow : newList.value
+    useEffect (
+      ()=>
+      {setFilteredDataToShow(IncludeFilter(newList.value,filterValue))},
+        [newList.value,filterValue]
+        )
   
+    
+
   return (
     <StyledUl ismutable={props.ismutable} ismain={props.ismain}>
       <h2 style={style.h2}>{props.header}</h2>
-      <h4 style={style.h4}>{filteredDataToShow.filter(item => item.type===props.header).length} from {(allMethods.filter(item => item.type===props.header)).length} contains '<span style={style.counter}>{filterValue}</span>'</h4>
-    <Input action={filterData} value={filterValue}/>
+      <h4 style={style.h4}>
+        {CompareFilter(filteredDataToShow,props.header).length} from {CompareFilter(newList.value, props.header).length} contains '<span style={style.counter}>{filterValue}</span>'
+      </h4>
+
+      <Input action={filterData} value={filterValue}/>
+
     <StyledResetButton onClick={resetFilter}>RESET FILTER</StyledResetButton>
-      {filteredData.filter(item => item.type===props.header).map(item => {
-        return <ListItem data = {item} key = {item.id} ismutable={props.ismutable} ismain={props.ismain} type={item.type} methodType={props.header}/>
+      { listTorender.filter(item => item.type===props.header).map(item => {
+        return <ListItem 
+        data = {item} 
+        key = {item.id} 
+        ismutable={props.ismutable} 
+        ismain={props.ismain} 
```

Comment:
can list be mutable & main in one time?

Reply:
'isMain' property created to define source list in styles